src/mozilla_taskgraph/worker_types.py

```diff
@@ -125,3 +127,60 @@ def build_shipit_payload(config, task, task_def):
     worker = task["worker"]
     task_def["tags"]["worker-implementation"] = "scriptworker"
     task_def["payload"] = {"release_name": worker["release-name"]}
+
+
+@payload_builder(
+    "scriptworker-signing",
+    schema={
+        Required("signing-type"): str,
+        # list of artifact URLs for the artifacts that should be signed
+        Required("upstream-artifacts"): [
+            {
+                # taskId of the task with the artifact
+                Required("taskId"): taskref_or_string,
+                # type of signing task (for CoT)
+                Required("taskType"): str,
+                # Paths to the artifacts to sign
+                Required("paths"): [str],
+                # Signing formats to use on each of the paths
+                Required("formats"): [str],
+                # Only For MSI, optional for the signed Installer
+                Optional("authenticode_comment"): str,
+            }
+        ],
+        Optional("max-run-time"): int,
+    },
+)
+def build_signing_payload(config, task, task_def):
+    worker = task["worker"]
+
+    task_def["payload"] = {
+        "upstreamArtifacts": worker["upstream-artifacts"],
+    }
+    if "max-run-time" in worker:
+        task_def["payload"]["maxRunTime"] = worker["max-run-time"]
+
+    task_def.setdefault("tags", {})["worker-implementation"] = "scriptworker"
+
+    formats = set()
+    for artifacts in worker["upstream-artifacts"]:
+        formats.update(artifacts["formats"])
+
+    scope_prefix = config.graph_config["scriptworker"]["scope-prefix"]
+    scopes = set(task_def.get("scopes", []))
+    scopes.add(f"{scope_prefix}:signing:cert:{worker['signing-type']}")
+    scopes.update({f"{scope_prefix}:signing:format:{format}" for format in formats})
```

Comment:
This bit is new (the gecko signing code deals with scopes in the `signing.py` transform through indirection like https://searchfox.org/mozilla-central/rev/04a2c5317c0af560ed1689304498416c9c6c485a/taskcluster/gecko_taskgraph/transforms/signing.py#175-177.

Am I reading correctly that the intent here is to push the scope definitions up to the kinds (or at least, some other transform will add them to the `worker` block rather than directly adding them in the transform as they do now?).